iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergQueryCompactor.java

```diff
@@ -121,6 +121,7 @@ public boolean run(CompactorContext context) throws IOException, HiveException,
       throw e;
     } finally {
       sessionState.setCompaction(false);
+      sessionState.closeOnExit();
```

Comment:
shouldn't we set the hook when we create a session object?  i don't think that is the right place